src/c++/perf_analyzer/load_manager.cc

```diff
@@ -110,6 +110,40 @@ LoadManager::GetAccumulatedClientStat(cb::InferStat* contexts_stat)
   return cb::Error::Success;
 }
 
+uint64_t
+LoadManager::GetIdleTime()
+{
+  uint64_t total{0};
+  size_t num_active_threads = 0;
+  for (auto& thread_stat : threads_stat_) {
+    std::lock_guard<std::mutex> lock(thread_stat->mu_);
+    thread_stat->idle_timer.Restart();
```

Comment:
Why restart inside the `GetIdleTime()` function? Why not after? To keep it as close to the reading time as possible? Why not use locks or something, or pause the timers?

Reply:
I'm confused on the relationship between GetIdleTime, ResetIdleTime(), Reset(), and Restart().

Reply:
Restart is used at the end of a measurement window. It will make sure any active counters have their time counted (but will stay active if already active) before asking for the idle time 
Reset is used at the start of a new measurement window. It will make sure any active counters start at 0 (but will stay active if already active)